software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -72,12 +129,12 @@ public void restart() {
         throw new UnsupportedOperationException();
     }
 
-    protected boolean calculateServiceUp() {
-        boolean up = false;
-        for (Entity member : slaves.getMembers()) {
-            if (Boolean.TRUE.equals(member.getAttribute(SERVICE_UP))) up = true;
-        }
-        return up;
-    }
+    private boolean calculateServiceUp() {
```

Comment:
Is this the appropriate method for calculating `SERVICE_UP`?

Before it would return `true` if one or more slave nodes were up, with no consideration of master.

Now it requires master and all slaves to be up.  Perhaps that is too strict, so perhaps the correct way is somewhere in the middle?